realsense2_camera/src/realsense_node_factory.cpp

```diff
@@ -38,60 +38,61 @@ void RealSenseNodeFactory::notification_handler(const rs2::notification &n, int
     std::lock_guard<std::recursive_mutex> scopedLock(_device_lock);
     if (iteration != this->_device_iteration)
     {
-        ROS_ERROR_STREAM("notification: device iterations don't match... ignoring duplicate notification for Device on: " << _usb_port_id);
+        ROS_ERROR_STREAM("realsense_camera: Device " << _usb_port_id << " device iterations don't match... ignoring duplicate notification for Device");
         return;
     }
-    this->_device_iteration++;
-    if (n.get_category() != RS2_NOTIFICATION_CATEGORY_FRAMES_TIMEOUT)
+    if (n.get_category() != RS2_NOTIFICATION_CATEGORY_FRAMES_TIMEOUT && n.get_category() != RS2_NOTIFICATION_CATEGORY_UNKNOWN_ERROR)
     {
-        ROS_ERROR_STREAM("notification: received a notification that is not RS2_NOTIFICATION_CATEGORY_FRAMES_TIMEOUT. The category was: " <<
+        ROS_ERROR_STREAM("realsense_camera: Device " << _usb_port_id << " received a notification that does not require reset. The category was: " <<
         rs2_notification_category_to_string(n.get_category()));
         return;
     }
-    std::thread([this]() {
-        ROS_ERROR_STREAM("notification: Executing reset for device " << _usb_port_id);
-        ROS_ERROR_STREAM("notification: Device " << _usb_port_id << " sleeping for 1 second");
+    this->_device_iteration++;
```

Comment:
Okay, so we *were* ignoring any subsequent notification after ignoring. I saw suspicious logs on bender that made me curious. Glad you caught this.